### PR TITLE
fix(contract): prevent ETH drainage vulnerability in tipping

### PR DESCRIPTION
**Pull request by shuhuiluo**
### Description

Fix ETH draining in spaces through tipping.

### Changes

- Add `msg.value` validation for native token tips to prevent attackers from draining contract ETH
- Add `UnexpectedETH` validation for ERC20 tips to prevent accidental ETH loss
- Add `MsgValueMismatch` and `UnexpectedETH` error types to ITipping interface
- Refactor imports to use relative paths and follow project conventions
- Use `CustomRevert` syntactic sugar throughout `TippingFacet`
- Add comprehensive test coverage for `msg.value` validation edge cases
- Combine duplicate conditional blocks for improved code efficiency

### Checklist

- [x] Tests added where required
- [x] Documentation updated where applicable
- [x] Changes adhere to the repository's contribution guidelines

### CODE REVIEW

File: packages/contracts/test/spaces/tipping/Tipping.t.sol [hunk 1]
```diff
@@ -234,4 +234,73 @@
...
+            TipRequest({
+                receiver: receiver,
+                tokenId: tokenId,
+                currency: CurrencyTransfer.NATIVE_TOKEN,
+                amount: amount,
+                messageId: messageId,
+                channelId: channelId
+            })
+        );
+    }
```
[the remark this answers is not in the document]
Comment (shuhuiluo) [reply packages/contracts/test/spaces/tipping/Tipping.t.sol:275]: The cleanest way is to use `vm.assume(amount != value);`.

File: packages/contracts/test/spaces/tipping/Tipping.t.sol [hunk 2]
```diff
@@ -234,4 +234,73 @@
...
+                receiver: receiver,
+                tokenId: tokenId,
+                currency: address(mockERC20),
+                amount: amount,
+                messageId: messageId,
+                channelId: channelId
+            })
+        );
+        vm.stopPrank();
+    }
```
[the remark this answers is not in the document]
Comment (shuhuiluo) [reply packages/contracts/test/spaces/tipping/Tipping.t.sol:305]: If a call reverts, all states must be reverted.
